core/src/main/scala/org/apache/spark/TaskContext.scala

```diff
@@ -150,6 +150,11 @@ abstract class TaskContext extends Serializable {
    */
   def stageId(): Int
 
+  /**
+   * An ID that is unique to the stage attempt that this task belongs to.
+   */
+  def stageAttemptId(): Int
```

Comment:
I think we should call it `stageAttempNumber` to be consistent with `taskAttemptNumber`. Also let's follow the comment of `attemptNumber`

Reply:
Yeah, if we are defining `stageAttemptId` from scratch, I would go for `stageAttemptNumber`.  However `stageAttemptId` are already used elsewhere in the codebase, Like in [Task.scala](https://github.com/apache/spark/blob/ded6d27e4eb02e4530015a95794e6ed0586faaa7/core/src/main/scala/org/apache/spark/scheduler/Task.scala#L55). I think it's more important to be consistent.

However I could update the comment to reflect the attempt number part if you wish